integration/request-test.ts

```diff
@@ -0,0 +1,162 @@
+              <pre id="loader-data">{JSON.stringify(loaderData)}</pre>
+              {actionData ?
+                <pre id="action-data">{JSON.stringify(actionData)}</pre> :
+                null}
+            </div>
+          )
+        }
+      `,
+    },
+  });
+
+  appFixture = await createAppFixture(fixture);
+});
+
+test.afterAll(() => appFixture.close());
+
+test("loader request on SSR GET requests", async ({ page }) => {
+  let app = new PlaywrightFixture(appFixture, page);
+  await app.goto("/");
+  await app.clickElement("#set-cookie");
+
+  let loaderData = JSON.parse(await page.locator("#loader-data").innerHTML());
+  expect(loaderData.method).toEqual("GET");
+  expect(loaderData.url).toMatch(/^http:\/\/localhost:\d+\/$/);
+  expect(loaderData.headers.cookie).toEqual(undefined);
+  expect(loaderData.text).toEqual("");
+
+  await app.clickElement("#submit-get-ssr");
+
+  loaderData = JSON.parse(await page.locator("#loader-data").innerHTML());
+  expect(loaderData.method).toEqual("GET");
+  expect(loaderData.url).toMatch(/^http:\/\/localhost:\d+\/\?type=ssr$/);
+  expect(loaderData.headers.cookie).toEqual("cookie=nomnom");
+  expect(loaderData.text).toEqual("");
+});
+
+test("loader request on CSR GET requests", async ({ page }) => {
+  let app = new PlaywrightFixture(appFixture, page);
+  await app.goto("/");
+  await app.clickElement("#set-cookie");
+
+  let loaderData = JSON.parse(await page.locator("#loader-data").innerHTML());
+  expect(loaderData.method).toEqual("GET");
+  expect(loaderData.url).toMatch(/^http:\/\/localhost:\d+\/$/);
+  expect(loaderData.headers.cookie).toEqual(undefined);
+  expect(loaderData.text).toEqual("");
+
+  await app.clickElement("#submit-get-csr");
+
+  loaderData = JSON.parse(await page.locator("#loader-data").innerHTML());
+  expect(loaderData.method).toEqual("GET");
+  expect(loaderData.url).toMatch(/^http:\/\/localhost:\d+\/\?type=csr$/);
+  expect(loaderData.headers.cookie).toEqual("cookie=nomnom");
+  expect(loaderData.text).toEqual("");
+});
+
+test("action + loader requests SSR POST requests", async ({ page }) => {
+  let app = new PlaywrightFixture(appFixture, page);
+  await app.goto("/");
+  await app.clickElement("#set-cookie");
+
+  let loaderData = JSON.parse(await page.locator("#loader-data").innerHTML());
+  expect(loaderData.method).toEqual("GET");
+  expect(loaderData.url).toMatch(/^http:\/\/localhost:\d+\/$/);
+  expect(loaderData.headers.cookie).toEqual(undefined);
+  expect(loaderData.text).toEqual("");
+
+  await app.clickElement("#submit-post-ssr");
+
+  let actionData = JSON.parse(await page.locator("#action-data").innerHTML());
+  expect(actionData.method).toEqual("POST");
+  expect(actionData.url).toMatch(/^http:\/\/localhost:\d+\/$/);
+  expect(actionData.headers.cookie).toEqual("cookie=nomnom");
+  expect(actionData.text).toEqual("type=ssr");
+
+  loaderData = JSON.parse(await page.locator("#loader-data").innerHTML());
+  expect(loaderData.method).toEqual("GET");
```

Comment:
Loaders receive `method:GET` on SSR "revalidations" now.  This fixes a bug in Remix 1.7.6 where the loader `request` was `POST` but without a `body`, and thus made for a different DX if you did a SSR/CSR submissions since `request.method` would be different in the `loader`.